src/main/tut/docs/module2.md

```diff
@@ -2,8 +2,398 @@
 layout: docs
 title: "Essentials"
 section: "docs"
-author: author
+author: Mauricio Saavedra
 ---
 
 # Module 2: Essentials
 
+## Scala expressions, types and values
+
+In functional programming an expression is a unit of code that returns a value after 
+executed. Expressions make a foundation of functional programming since they allow to
+return data instead of modifying existing data.
+
+An expression is as simple as:
+
+```scala
+"hello world"
+1 + 1
+5 < 3
+``` 
+
+### Variables
+
+In this context, we can define a variable as the result of an expression. In Scala,
+you can define variables in three basic ways.
+We will further view in detail its differences
+and usages, but for now lets settle with only learning how to define them:
+
+#### val
+`val name: type = value`
+ 
+For example:
+```scala
+val number: Int = 5
+val word: String = "Hi!"
+val something = 45.6
+```
+
+A variable defined with `val` will be immutable, the type annotation is optional, but

```

Comment:
val does not imply immutability per se. be carefull here. the hole value will be immutable but maybe not the elements of it if its a mutable data strucutre.